frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkLoadManager.java

```diff
@@ -112,7 +116,20 @@ public int getNumRunningWorkers(String modelName) {
                 for (int i = currentWorkers - 1; i >= maxWorker; --i) {
                     // TODO: kill unhealthy worker first.
                     WorkerThread thread = threads.remove(i);
+                    WorkerLifeCycle lifecycle = thread.getLifeCycle();
                     thread.shutdown();
+
+                    Process workerProcess = lifecycle.getProcess();
+
+                    if (workerProcess.isAlive()) {
+                        workerProcess.destroyForcibly();
+                        try {
+                            workerProcess.waitFor();
```

Comment:
Look into if we can wait for some amount of time. This otherwise seems to wait for ever (until the process finishes). [waitFor reference](https://docs.oracle.com/javase/8/docs/api/java/lang/Process.html)